__tests__/app.test.js

```diff
@@ -85,36 +87,114 @@ describe("GET /api/articles/:article_id", () => {
+                    expect(typeof article.comment_count).toBe('number');
+                    
+                })
+            });
+    })
+    
+    describe('error handling', () => {
+      
+        test('responds with a 404 status when given a valid but non-existent endpoint', () => {
+            return request(app)
+                .get('/api/article')
+                .expect(404)
+        })
+       
     })
 })
 
+//// question 6 getCommentsByArticle
+    
+
+describe("GET '/api/articles/:article_id/comments'", () => {
+    test('GET:commentsByArticleId should return comments associated with the passed article sorted by most recent first', () => {
+        return request(app)
+            .get('/api/articles/1/comments')
+            .expect(200)
+            .then((response) => {
+                const comments = response.body
+                    
+                expect(Array.isArray(comments)).toBe(true)
+                expect(comments).toBeSortedBy('created_at', { descending: true, })
+                expect(comments.length).toBe(11)
+
+
+                comments.forEach((comment) => {
+                    expect(comment).toHaveProperty("comment_id")
+                    expect(typeof comment.comment_id).toBe('number');
+                    expect(comment).toHaveProperty("votes")
+                    expect(typeof comment.votes).toBe('number');
+                    expect(comment).toHaveProperty("created_at")
+                    expect(typeof comment.created_at).toBe('string');
+                    expect(comment).toHaveProperty("author")
+                    expect(typeof comment.author).toBe('string');
+                    expect(comment).toHaveProperty("body")
+                    expect(typeof comment.body).toBe('string');
+                    expect(comment).toHaveProperty("article_id")
+                    expect(typeof comment.article_id).toBe('number');
+                })
+            })
+    })
+
+    describe('edge cases', () => {
+        test('responds with an empty array if there are no comments for the specified article', () => {
+            
+            return request(app)
+                .get('/api/articles/7/comments')
+                .expect(200)
+                .then((response) => {
+                    const comments = response.body
+                    expect(comments).toEqual([])
+
+                })
+        })
+    })
+    describe('error handling', () => {
+        test('responds with a 400 status when given an invalid endpoint', () => {
+            return request(app)
+                .get('/api/articles/forklift/comments')
+                .expect(400)
+        })
+    })
+     describe('error handling', () => {
+        test('responds with a 404 status when given a valid but non-existent endpoint', () => {
+            return request(app)
+                .get('/api/articles/forklift/comment')
+                .expect(404)
+        })
+    })
+})
```

Comment:
This test is not quite right and does not match its description. A valid but non-existent id would be something like `99999` not `forklift` (which is a repeat of the correct 400 test above). Please change this test then update your code to pass. 🙂